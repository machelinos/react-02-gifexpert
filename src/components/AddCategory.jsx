import { useState } from "react";

export const AddCategory = ({onAddCategory}) => {
    const [category, setCategory] = useState('');

    const handleSubmit = (event) => {
        event.preventDefault();
        const categoryToEmmit = category.trim();
        if(category.trim().length <= 1) return;
        onAddCategory(categoryToEmmit);
        setCategory('');
    };

    const handleCategoryChange = (event)=>{
        setCategory(event.target.value);
    }

    return (
        <form
            onSubmit={handleSubmit}
        >
            <input
                onChange={handleCategoryChange}
                placeholder="Enter category" 
                type="text"
                value={category}
            />

            <button
                type="submit"
            >Add category</button>

        </form>
    )
}
