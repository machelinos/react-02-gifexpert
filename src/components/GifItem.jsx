import PropTypes from "prop-types";

export const GifItem = ({title, id, imgUrl}) => {
  return (
    <div className="card">
        <img src={imgUrl} alt={title} />
        <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  imgUrl: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
}
