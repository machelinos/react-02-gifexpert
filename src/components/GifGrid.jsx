import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    const { gifs, isLoading } = useFetchGifs(category);

    return (
        <>
            <h2>{ category }</h2>
            { isLoading && (<h3>Loading...</h3>) }

            <div className="card-grid">
                {
                    gifs.map((gif)=>(
                        <GifItem
                            key={gif.id}
                            {...gif}
                        />
                    ))
                }
            </div>
        
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
