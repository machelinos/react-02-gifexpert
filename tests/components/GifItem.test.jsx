import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', ()=>{
    const title="Image title";
    const id = "123456";
    const imgUrl = "https://title.com/imahe.jpg";

    test('debe hacer match con el snapshot', ()=>{
        const {container} = render(<GifItem title={title} id={id} imgUrl={imgUrl} />);
        expect(container).toMatchSnapshot();
    });

    test('image src and alt text must match', ()=>{
        render(<GifItem title={title} id={id} imgUrl={imgUrl} />);
        // screen.debug();
        const {src, alt } = screen.getByRole('img');
        expect(src).toBe(imgUrl);
        expect(alt).toBe(title);
    });

    test('test title should be present in the component rendered',()=>{
        render(<GifItem title={title} id={id} imgUrl={imgUrl} />);
        const text = screen.getByText(title);
        expect(text).toBeTruthy();
    });
})