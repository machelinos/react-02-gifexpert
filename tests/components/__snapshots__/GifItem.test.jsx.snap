// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Image title"
      src="https://title.com/imahe.jpg"
    />
    <p>
      Image title
    </p>
  </div>
</div>
`;
